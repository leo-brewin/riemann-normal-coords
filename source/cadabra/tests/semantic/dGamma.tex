\documentclass[12pt]{cdblatex}
\usepackage{fancyhdr}
\usepackage{footer}

\begin{document}

\section*{\jobname}

\CdbSetup{action=hide}

\begin{cadabra}
   import shared

   import cdblib

   term00A = cdblib.get ('check000','expected/dGamma.json')
   term01A = cdblib.get ('check001','expected/dGamma.json')
   term02A = cdblib.get ('check002','expected/dGamma.json')
   term03A = cdblib.get ('check003','expected/dGamma.json')
   term04A = cdblib.get ('check004','expected/dGamma.json')

   term00B = cdblib.get ('check000','output/dGamma.json')
   term01B = cdblib.get ('check001','output/dGamma.json')
   term02B = cdblib.get ('check002','output/dGamma.json')
   term03B = cdblib.get ('check003','output/dGamma.json')
   term04B = cdblib.get ('check004','output/dGamma.json')

   diff000 = shared.check (term00A,term00B)   # cdb (diff000,diff000)
   diff001 = shared.check (term01A,term01B)   # cdb (diff001,diff001)
   diff002 = shared.check (term02A,term02B)   # cdb (diff002,diff002)
   diff003 = shared.check (term03A,term03B)   # cdb (diff003,diff003)
   diff004 = shared.check (term04A,term04B)   # cdb (diff004,diff004)

\end{cadabra}

\begin{dgroup*}
   \Dmath*{ \cdb*{diff000} }
   \Dmath*{ \cdb*{diff001} }
   \Dmath*{ \cdb*{diff002} }
   \Dmath*{ \cdb*{diff003} }
   \Dmath*{ \cdb*{diff004} }
\end{dgroup*}

\end{document}
