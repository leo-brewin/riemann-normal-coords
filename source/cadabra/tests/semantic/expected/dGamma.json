{
  "check000": "R^{a}_{b c d}",
  "check001": "3\\partial_{e}(R^{a}_{b c d})",
  "check002": "9\\partial_{f e}(R^{a}_{b c d})-R^{a}_{c e g} R^{g}_{b f d}-R^{a}_{c f g} R^{g}_{b e d}",
  "check003": "6\\partial_{g f e}(R^{a}_{b c d})-R^{h}_{b g d} \\partial_{e}(R^{a}_{c f h})-R^{h}_{b f d} \\partial_{e}(R^{a}_{c g h})-R^{a}_{c e h} \\partial_{f}(R^{h}_{b g d})-R^{h}_{b e d} \\partial_{f}(R^{a}_{c g h})-R^{a}_{c f h} \\partial_{e}(R^{h}_{b g d})-R^{a}_{c g h} \\partial_{e}(R^{h}_{b f d})",
  "check004": "180\\partial_{h g f e}(R^{a}_{b c d})-36R^{i}_{b g d} \\partial_{h e}(R^{a}_{c f i}) + 4R^{a}_{f e i} R^{i}_{c h j} R^{j}_{b g d} + 4R^{a}_{f h i} R^{i}_{c e j} R^{j}_{b g d}-72R^{i}_{b f d} \\partial_{h e}(R^{a}_{c g i}) + 8R^{a}_{g e i} R^{i}_{c h j} R^{j}_{b f d} + 8R^{a}_{g h i} R^{i}_{c e j} R^{j}_{b f d}-45\\partial_{e}(R^{a}_{c f i}) \\partial_{g}(R^{i}_{b h d})-45\\partial_{e}(R^{a}_{c g i}) \\partial_{f}(R^{i}_{b h d})-45\\partial_{e}(R^{a}_{c h i}) \\partial_{f}(R^{i}_{b g d})-36R^{a}_{c e i} \\partial_{h f}(R^{i}_{b g d}) + 4R^{a}_{c e i} R^{i}_{g f j} R^{j}_{b h d} + 4R^{a}_{c e i} R^{i}_{g h j} R^{j}_{b f d}-36R^{i}_{b e d} \\partial_{h f}(R^{a}_{c g i}) + 4R^{a}_{g f i} R^{i}_{c h j} R^{j}_{b e d} + 4R^{a}_{g h i} R^{i}_{c f j} R^{j}_{b e d}-45\\partial_{f}(R^{a}_{c g i}) \\partial_{e}(R^{i}_{b h d})-45\\partial_{f}(R^{a}_{c h i}) \\partial_{e}(R^{i}_{b g d})-72R^{a}_{c f i} \\partial_{h e}(R^{i}_{b g d}) + 8R^{a}_{c f i} R^{i}_{g e j} R^{j}_{b h d} + 8R^{a}_{c f i} R^{i}_{g h j} R^{j}_{b e d}-45\\partial_{g}(R^{a}_{c h i}) \\partial_{e}(R^{i}_{b f d})-36R^{a}_{c g i} \\partial_{h e}(R^{i}_{b f d}) + 4R^{a}_{c g i} R^{i}_{f e j} R^{j}_{b h d} + 4R^{a}_{c g i} R^{i}_{f h j} R^{j}_{b e d}"
}
